In [6]:
from __future__ import absolute_import, division, print_function, unicode_literals

# Install TensorFlow
try:
  %tensorflow_version 2.x
except Exception:
  pass

import tensorflow as tf
print(tf.__version__)

2.0.0


Load and prepare the [MNIST dataset](http://yann.lecun.com/exdb/mnist/). Convert the samples from integers to floating-point numbers:

In [2]:
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

11493376/11490434 [==============================] - 5s 0us/step


Build the `tf.keras.Sequential` model by stacking layers. Choose an optimizer and loss function for training:

In [3]:
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10, activation='softmax')
])

model.compile(optimizer='adam',
              loss='sparse_categorical_crossentropy',
              metrics=['accuracy'])

Train and evaluate the model:

In [4]:
model.fit(x_train, y_train, epochs=5)

model.evaluate(x_test,  y_test, verbose=2)

Train on 60000 samples
Epoch 1/5
60000/60000 [==============================] - 10s 163us/sample - loss: 0.2948 - accuracy: 0.9145
Epoch 2/5
60000/60000 [==============================] - 5s 91us/sample - loss: 0.1413 - accuracy: 0.9578
Epoch 3/5
60000/60000 [==============================] - 5s 90us/sample - loss: 0.1061 - accuracy: 0.9670
Epoch 4/5
60000/60000 [==============================] - 6s 95us/sample - loss: 0.0854 - accuracy: 0.9734
Epoch 5/5
60000/60000 [==============================] - 5s 88us/sample - loss: 0.0723 - accuracy: 0.9769
10000/1 - 4s - loss: 0.0384 - accuracy: 0.9778


[0.07436289865709841, 0.9778]

The image classifier is now trained to ~98% accuracy on this dataset